### ISSUE TITLE
Questions about buffer writeback

### ISSUE THREAD

**Issue by Alkaid-Benetnash** [sjp38/linux#1]
Hi SeongJae,

Thanks for your work and I hope it can be accepted one day.
Here are some questions about buffer writeback ([damon_write_rbuf](https://github.com/sjp38/linux/blob/damon/patches/v18/mm/damon.c#L748)):
1. Is it an optimization if you only open the record file once during setup, instead of opening, closing it everytime?
2. I had some trouble when running damon for a long time (damon.data > 2GiB). I think you should add flag `O_LARGEFILE` in `filp_open`.

Thanks!

**Comment by sjp38** [comment]
So sorry for this late reply!  I use Github as code hosting purpose only and therefore just found your question.

> 1. Is it an optimization if you only open the record file once during setup, instead of opening, closing it everytime?

Agreed, it must be better.  I will make that kind of optimization in the next spin.

> 2. I had some trouble when running damon for a long time (damon.data > 2GiB). I think you should add flag O_LARGEFILE in filp_open.

Agreed, I will do so in the next spin.

Thanks for this valuable question!

Thanks,
SeongJae Park

**Comment by sjp38** [comment]
Nonetheless, I can't react on Github fast.  Please consider using the mailing list from next time.  I will add you as a recipient of the patchset posting.

**Comment by sjp38** [comment]
Hmm, I'm now unsure about the first question.  Maybe you could increase the buffer size if it is real overhead.  Have you tried that?

**Comment by sjp38** [comment]
If you clearly see the open()/close() overhead even after adjusting the buffer size, please let me know.  Unless, I will not make the change.

**Comment by sjp38** [comment]
The 2nd request is made: https://github.com/sjp38/linux/tree/damon/for-alkaid

However, it would take some time for me to set up a 32-bit test environment.  It would be great if you could test it and allowing me the result.  Could you please?

**Comment by Alkaid-Benetnash** [comment]
Hi,
These two requests are not big deals, I would continue commenting here.

For 1st request, I agree to not change anything. It is more of an instinct than real performance issues.
For 2nd request, I experiened file write error over 2GB files on a 64-bit environment (CONFIG_64BIT=y). I have added the flag myself and it solved 2GB file problems for me. Since you have already included this change in the new patchset, I would consider this issue solved.
Thank you for taking care of these.
